### PR TITLE
MF3-M04: fix(nitronode): fix uint8 overflow in quorum weight accumulators

### PR DESCRIPTION
**Pull request by nksazonov**
## Summary

- `totalWeights` in `CreateAppSession` and `achievedQuorum` in `verifyQuorum` were declared as `uint8`, causing silent modulo-256 wrap-around when participant weights summed above 255
- Widened both accumulators to `uint16` (sufficient for up to 32 participants × 255 weight = 8160)
- No external types changed — `SignatureWeight` and `Quorum` remain `uint8` at the wire/storage layer

## Test plan

- [ ] `TestCreateAppSession_TotalWeightsOver255` — creation succeeds when real total > 255 (previously false-rejected)
- [ ] `TestCreateAppSession_TotalWeightsWrapToZero` — 128+128=256 wrap-to-zero boundary
- [ ] `TestCreateAppSession_QuorumExceedsRealTotalOver255_Rejected` — guard still rejects quorum > real total
- [ ] `TestSubmitAppState_VerifyQuorumWeightOver255` — valid coalition with combined weight > 255 passes
- [ ] `TestSubmitAppState_VerifyQuorumWrapToZero` — 128+128 boundary through SubmitAppState
- [ ] `TestSubmitDepositState_VerifyQuorumWeightOver255` — same coverage through SubmitDepositState
- [ ] `TestRebalanceAppSessions_VerifyQuorumWeightOver255` — same coverage through RebalanceAppSessions

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed quorum verification to correctly handle participant signature weights exceeding 255, ensuring app sessions are properly validated regardless of total weight magnitude.

* **Tests**
  * Added comprehensive test coverage for quorum verification edge cases and weight overflow scenarios across multiple operations.

### CODE REVIEW

**Comment by nksazonov** [comment]
@coderabbitai review

File: nitronode/api/app_session_v1/create_app_session.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func (h *Handler) CreateAppSession(c *rpc.Context) {
 	// derivation, and stored participant list all see the canonical (lowercase, 0x-prefixed)
 	// representation regardless of how the caller cased the address or whether they
 	// included the 0x prefix.
-	var totalWeights uint8
+	var totalWeights uint16
```
Comment (philanton) [inline nitronode/api/app_session_v1/create_app_session.go:78]: `uint16` only holds the sum up to 257 participants (257×255 = 65535). `NITRONODE_MAX_PARTICIPANTS` is a plain `int`, default 32, with no upper bound — set it ≥ 258 and `totalWeights` wraps again, same bug class. This is the second half of the finding's recommendation ("use such maxParticipants value that the maximum weight fits uint16"), and it's not enforced anywhere. Default-safe + operator-only, so low severity, but I'd add a startup validation that `MaxParticipants` ≤ 257 (or just ≤ 32) over in `runtime.go`/`main.go` so the bound can't drift. Cheap to close and it's exactly what the finding asked for.

File: nitronode/api/app_session_v1/create_app_session_test.go [hunk 2]
```diff
@@ -1231,3 +1313,139 @@ func TestCreateAppSession_DuplicateParticipantAcrossCases(t *testing.T) {
...
+		t.Fatalf("total weights 256 (128+128) with quorum 128 must be accepted, got: %v", respErr)
+	}
+	assert.Equal(t, rpc.MsgTypeResp, ctx.Response.Type)
+	mockStore.AssertExpectations(t)
+}
+
+// TestCreateAppSession_QuorumExceedsRealTotalOver255_Rejected verifies that when total
+// weights exceed 255, a quorum genuinely larger than the real total is still rejected.
+// (The fix must not disable the quorum>total guard for the >255 range.)
+func TestCreateAppSession_QuorumExceedsRealTotalOver255_Rejected(t *testing.T) {
```
Comment (philanton) [inline nitronode/api/app_session_v1/create_app_session_test.go]: The name says `Over255` but this uses total 100+100=200 — and as the inline comment admits, it can't actually test the >255 range because `Quorum` is `uint8` (capped at 255). I'd drop `Over255` from the name. Worth noting the underlying thing: since `Quorum` stays `uint8` while `totalWeights` can now reach 8160, the `quorum > totalWeights` guard can never fire once total ≥ 255 — not a bug, just dead in that range, and it means you can't require a threshold above 255 even with large weights. That's a pre-existing `uint8` `Quorum` limit, not something this PR introduces, but can you confirm it's intended?

**Review by philanton — COMMENTED** [review]
Nice job, fix is correct and tests cover the real path. Two notes inline.

**Review by ihsraham — COMMENTED** [review]
Looks close overall. The main accumulator fix is doing the right thing: `CreateAppSession` and shared `verifyQuorum` now sum in `uint16`, I did not find another production `SignatureWeight` quorum accumulator, and the real overflow regressions fail on the base branch and pass here across create, submit app state, submit deposit state, and rebalance.

The only thing I would keep open for M-04 closure is Anton's `MaxParticipants` note. The audit asked for the accumulator type and the participant bound to agree, so this should either reject `NITRONODE_MAX_PARTICIPANTS > 257` at startup or use a wider accumulator. The default/charts are at 32, so I do not see a default-path exploit, but I would still close that before marking the audit item fixed.

Tooling-wise, Snyk is green and there are no manifest changes in this PR. Local `go test ./...`, `go test -race ./nitronode/api/app_session_v1`, `go vet ./nitronode/...`, `go mod verify`, `gofmt -l`, and `git diff --check` were clean.

File: nitronode/api/app_session_v1/create_app_session.go (hunk 1 above)
Comment (nksazonov) [reply nitronode/api/app_session_v1/create_app_session.go:78]: 36fa562f

File: nitronode/api/app_session_v1/create_app_session_test.go (hunk 2 above)
Comment (nksazonov) [reply nitronode/api/app_session_v1/create_app_session_test.go]: 36fa562f
